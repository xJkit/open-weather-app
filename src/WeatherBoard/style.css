.weather-board {
  text-align: center;
}

.weather-table {
  display: inline-block;
  margin-top: 32px;
}