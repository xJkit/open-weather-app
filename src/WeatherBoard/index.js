import React from 'react';
import './style.css';

export default ({ temp, temp_max, temp_min, humidity, clouds, description, forecastTime, name }) => {
  return (
    <div className="weather-board">
      <table className="weather-table">
        <thead>
          <tr>
            <th className="title">forecastTime: {forecastTime}</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>城市： {name || '沒有資料'}</td>
          </tr>
          <tr>
            <td>溫度： {temp ? `${temp} 度` : '沒有資料'}</td>
          </tr>
          <tr>
            <td>最高溫度： {temp_max ? `${temp_max} 度` : '沒有資料'}</td>
          </tr>
          <tr>
            <td>最低溫度： {temp_min ? `${temp_min} 度` : '沒有資料'}</td>
          </tr>
          <tr>
            <td>濕度： {humidity ? `${humidity}%` : '沒有資料'}</td>
          </tr>
          <tr>
            <td>Clouds: {clouds ? `${clouds}% (${description})` : '沒有資料'}</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}