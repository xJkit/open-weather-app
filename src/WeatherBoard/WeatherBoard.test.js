import React from 'react';
import renderer from 'react-test-renderer';
import WeatherBoard from './index';

describe('WeatherBoard', () => {
  test('sanity check', () => {
    expect(1).toBe(1);
  });

  test('renders correctly', () => {
    const tree = renderer.create(
      <WeatherBoard />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
