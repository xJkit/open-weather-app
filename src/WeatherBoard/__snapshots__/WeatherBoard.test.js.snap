// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherBoard renders correctly 1`] = `
<div
  className="weather-board"
>
  <table
    className="weather-table"
  >
    <thead>
      <tr>
        <th
          className="title"
        >
          forecastTime: 
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          城市： 
          沒有資料
        </td>
      </tr>
      <tr>
        <td>
          溫度： 
          沒有資料
        </td>
      </tr>
      <tr>
        <td>
          最高溫度： 
          沒有資料
        </td>
      </tr>
      <tr>
        <td>
          最低溫度： 
          沒有資料
        </td>
      </tr>
      <tr>
        <td>
          濕度： 
          沒有資料
        </td>
      </tr>
      <tr>
        <td>
          Clouds: 
          沒有資料
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
